### DISCUSSION TITLE
How to dynamically hide column in react-table?

### DISCUSSION THREAD

**Discussion by dhanushkumarsivaji** [TanStack/table#4228] [Q&A]
I have created the react-table. Here am trying to hide the column dynamically. It means if the data is missing in the API for a particular column, I need to hide the column entirely.

I tried by giving the property ```show: false```. But it seems this property is not working in the version of ```@tanstack/react-table``` anymore.

Here you can get the working demo link of the table - https://codesandbox.io/s/tanstack-sorting-9dp6o0?file=/src/main.tsx:208-229

Is there any way I can hide the column dynamically in react-table. Please let me know the possibility of that

**Comment by simPod** [comment]
You can set `initialState: { columnVisibility: { firstName: false } }`
